overlord/snapstate/conflict.go

```diff
@@ -71,7 +71,7 @@ func AddAffectedSnapsByKind(kind string, f AffectedSnapsFunc) {
 
 func affectedSnaps(t *state.Task) ([]string, error) {
 	// snapstate's own styled tasks
-	if t.Has("snap-setup") || t.Has("snap-setup-task") {
+	if t.Has("snap-setup") || t.Has("snap-setup-task") || t.Has("restore-state") {
```

Comment:
We could avoid special-casing it here if "snap-setup" was added to the restore-state task, wdyt? This would not handle tasks created by old snapd but that is fine I think given that this is a short-lived task (and the bug is in edge case).

Reply:
It sounds like a reasonable suggestion to me, let's wait for Samuele's opinion.

Reply:
I reverted this change, it turns out it was not needed :-)